torch/csrc/jit/passes/onnx/function_extraction.cpp

```diff
@@ -0,0 +1,1101 @@
+  val_attr_to_name_[v_it->second] = name;
+}
+
+c10::optional<std::string> FunctionExtractor::FunctionContext::FindAttrName(
+    Node* ref_n,
+    Symbol attr) {
+  auto v_it =
+      scope_ctxs_[scope_key_]->env_to_subgraph_.find(ref_n->outputs().at(0));
+  if (v_it == scope_ctxs_[scope_key_]->env_to_subgraph_.end()) {
+    return c10::nullopt;
+  }
+  auto* n_in_def = v_it->second->node();
+  auto n_attr_it = node_attr_to_name_.find(n_in_def);
+  if (n_attr_it == node_attr_to_name_.end()) {
+    return c10::nullopt;
+  }
+  auto name_it = n_attr_it->second.find(attr.toUnqualString());
+  if (name_it == n_attr_it->second.end()) {
+    return c10::nullopt;
+  }
+  return name_it->second;
+}
+
+c10::optional<std::string> FunctionExtractor::FunctionContext::FindAttrName(
+    Node* ref_const_n) {
+  TORCH_INTERNAL_ASSERT(ref_const_n->kind() == c10::onnx::Constant);
+  auto v_it =
+      scope_ctxs_[scope_key_]->env_to_subgraph_.find(ref_const_n->output());
+  if (v_it == scope_ctxs_[scope_key_]->env_to_subgraph_.end()) {
+    return c10::nullopt;
+  }
+  auto name_it = val_attr_to_name_.find(v_it->second);
+  if (name_it == val_attr_to_name_.end()) {
+    return c10::nullopt;
+  }
+  return name_it->second;
+}
+
+void FunctionExtractor::DebugPrintScopeContexts(
+    const scope_ctx_map& scope_ctxs) {
+  for (auto& it : scope_ctxs) {
+    GRAPH_UPDATE(
+        "Scope name: ",
+        it.first->namesFromRoot(),
+        " ",
+        it.first->name().toDisplayString());
+    GRAPH_UPDATE("Children scopes: ", [&]() {
+      std::stringstream ss;
+      for (const auto& child_scope : it.second->children_) {
+        ss << child_scope->name().toDisplayString() << " ";
+      }
+      return ss.str();
+    }());
+    GRAPH_UPDATE("Node types: \n", [&]() {
+      std::stringstream ss;
+      for (auto n : it.second->nlist_) {
+        ss << "  " << *n;
+      }
+      return ss.str();
+    }());
+    GRAPH_UPDATE("Node count: ", it.second->nlist_.size());
+  }
+}
+
+void FunctionExtractor::DebugPrintGraphWithFunction(
+    const std::shared_ptr<Graph>& g) {
+  GRAPH_UPDATE("Local function definitions:");
+  for (auto* n : g->nodes()) {
+    if (n->kind() == Symbol::onnx("LocalFunctionDef")) {
+      GRAPH_UPDATE(
+          n->s(attr::name),
+          " graph: ",
+          n->g(Symbol::attr("graph"))->toString());
+    }
+  }
+  GRAPH_UPDATE("Main graph: ", g->toString());
+}
+
+bool FunctionExtractor::IsValidScope(ScopePtr s) {
+  return !s->isRoot() && !s->isBlank();
+}
+
+bool FunctionExtractor::IsAncester(ScopePtr parent, ScopePtr child) {
+  if (!IsValidScope(parent) || !IsValidScope(child) ||
+      parent->getDepth() >= child->getDepth()) {
+    return false;
+  }
+  do {
+    child = child->parent();
+    if (parent == child) {
+      return true;
+    }
+  } while (IsValidScope(child));
+  return false;
+}
+
+c10::optional<ScopePtr> FunctionExtractor::FindCommonAncestor(
+    ScopePtr a,
+    ScopePtr b) {
+  auto common_ancestor = a;
```

Comment:
It's an unuse variable in this function?

Reply:
thx, removed